OVERVIEW.md

```diff
@@ -0,0 +1,102 @@
+
+
+## Flow
+
+### Webhooks
+- Webhook, important to check secret
+ - Listens for pull_request events of type 'opened', 'closed', and 'synchronised'.
+- ...
+- Starts ./run demo
+
+_**When receiving webhooks, it is important to always check the secret hash. ALWAYS.**_
+
+#### Github webhooks
+Github webhooks are pushed to the `/webhook/github` path. When the app receives a payload from GitHub it will check the signature against a hash made from our secret. We must check this to ensure that only webhooks from verified sources are started.
+
+Github can send a variety of event types. The type is checked against the header `HTTP_X_GITHUB_EVENT`. This then checks the type of event in the payload content. We primarily listen for `opened`, `closed`, and `synchronized`. Synchronized is triggered when a branch has been updated; often by rebasing or pushing new commits.
+
+When the webhook has been understood, it queues up a relevant task in the message queue. It will either start/update a demo or delete one. Starting a demo will stop any previously running demos, so this action can be used for updating a demo.
+
+### Frontend management
+The demo service has a simple admin/management interface for viewing demos, as well as starting and stopping them. This interface requires an openid login and this should be enforced to prevent unauthorised users creating demos on the system.
+
+Running demos are displayed by filtering the Docker containers on the machine with the relevant `run.demo=true` label.
+
+### Running tasks
+- Task chain
+- Retries a few times with some back off
+- When worker has space, runs next task in queue
+- Set labels
+
+Tasks are created as a [Celery chain](http://docs.celeryproject.org/en/latest/userguide/canvas.html#chains). This allows multiple smaller tasks to run in sequence and act independantly. For example, the step to notify GitHub is run as a seperate task which can retry without the overhead of restarting the whole demo. This is particularly beneficial for steps involving network requests as they can fail because of short lived network errors.
+
+The tasks are added to the RabbitMQ service through Celery. When a Celery worker is free it will pick up the latest task of the top of the queue. If a task fails, they are configured to retry multiple times with an exponential back off. The first failure will wait 2 seconds, the seconds attempt will wait 4 seconds, then 8 seconds, then 16 seconds and so on.
+
+New demos are run by downloading the source code to a `/srv/demos` subfolder and running `./run serve --detached` in this folder. Docker labels are used to add metadata and manage the demos. It is beneficial to add more data than you need as it is harder to add later without restarting demos.
+
+These are a list of labels which are added for the demo system:
+- `run.demo=true`
+- `run.demo.url`
+- `run.demo.url_full` - Full URL for links
+- `run.demo.github_user`
+- `run.demo.github_repo`
+- `run.demo.github_pr`
```

Comment:
I know it's understandable but could you add description for every label. Because I am not sure the difference between
```
run.demo.url
run.demo.url_full
```